WPMedia/fusion-news-theme-blocks
# Small Improvement:  Prevent direct state management which could cause issues

## Description

Fixed potential issue in react

Fixes https://lgtm.com/projects/g/WPMedia/arc-themes-blocks/?mode=tree&ruleFocus=1823611029

## Jira Ticket

- improvement poc. could be unstable and bad practice

## Acceptance Criteria

- Removes the warnings for direct state 
- Still updates as expected on click 

## Test Steps

1. Checkout this branch `git checkout REFACTOR-change-direct-state-mutation`
2. Run fusion repo with linked blocks `npx fusion start -f -l `
3. Go to http://localhost/pf/search/video/?_website=the-sun
4. Click show more button at the bottom. See new page created as before 
<img width="865" alt="Screen Shot 2022-03-11 at 10 30 01" src="https://user-images.githubusercontent.com/5950956/157908171-feb8c742-8a01-491c-847e-642ac194c30e.png">

## Effect Of Changes

### Before

- Direct react set state warnings

### After

- No direct react set state warnings


## Dependencies or Side Effects

- Although there's still a setState in the constructor which is bad practice

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored). -> I'm not sure if adding test coverage here would be feasible or helpful without a significant refactor

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
